config/credentials.go

```diff
@@ -0,0 +1,424 @@
+//  2. Atomically swaps the ..data symlink to point to the new directory
+//  3. Deletes the old timestamped directory
+//
+// The individual file symlinks (token, ca-bundle.crt) don't change - only ..data changes.
+// This is why the manager watches for ..data changes to detect secret rotation.
+//
+// Certificate Pool:
+//   - Call InitializeCertPool() to set up system CAs plus custom CA bundles
+//   - Use GetCertPool() to get a clone of the managed certificate pool
+//   - CA bundle files are watched and the pool is rebuilt automatically on changes
+//
+// Usage:
+//
+//	token, err := conf.Credentials.Get(conf.ExternalServices.Prometheus.Auth.Token)
+//	certPool := conf.Credentials.GetCertPool()
+//
+// Internal Data Structures:
+//   - cache: maps absolute file path → credential content (trimmed string)
+//   - watchedDirs: maps directory path → unused struct{} (used as a set of watched directories)
+type CredentialManager struct {
+	mu          sync.RWMutex
+	cache       map[string]string
+	watchedDirs map[string]struct{}
+	watcher     *fsnotify.Watcher
+	done        chan struct{}
+	closeOnce   sync.Once
+
+	// Certificate pool management
+	certPool      *x509.CertPool
+	caBundlePaths []string
+}
+
+// NewCredentialManager creates a new credential manager with file watching enabled.
+// Returns an error if the file watcher cannot be initialized.
+func NewCredentialManager() (*CredentialManager, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create credential file watcher: %w", err)
+	}
+
+	cm := &CredentialManager{
+		cache:       make(map[string]string),
+		watchedDirs: make(map[string]struct{}),
+		watcher:     watcher,
+		done:        make(chan struct{}),
+	}
+
+	go cm.watchFiles()
+	return cm, nil
+}
+
+// Close stops the file watcher and cleans up resources.
+// Should be called during application shutdown.
+func (cm *CredentialManager) Close() {
+	if cm == nil {
+		return
+	}
+	cm.closeOnce.Do(func() {
+		close(cm.done)
+		if cm.watcher != nil {
+			cm.watcher.Close()
+			// Note: Don't set cm.watcher = nil here
+			// The watchFiles goroutine may still be accessing cm.watcher.Events
+			// Closing the watcher closes its channels, causing watchFiles to exit gracefully
+		}
+	})
+}
+
+// InitializeCertPool sets up the certificate pool with system CAs and additional CA bundles.
+// This should be called once during configuration initialization.
+// The CA bundle files will be automatically watched and the pool rebuilt on changes.
+func (cm *CredentialManager) InitializeCertPool(caBundlePaths []string) error {
+	if cm == nil {
+		return fmt.Errorf("credential manager not initialized")
+	}
+
+	cm.mu.Lock()
+	cm.caBundlePaths = caBundlePaths
+	cm.mu.Unlock()
+
+	return cm.rebuildCertPool()
+}
+
+// rebuildCertPool rebuilds the certificate pool from system CAs and configured CA bundles.
+func (cm *CredentialManager) rebuildCertPool() error {
```

Comment:
This looks good and since we're already building the cert pool here I think it's further proof you don't need to do this again in `Config.CertPool()`.

Reply:
code changed a bunch since this comment. Config.CertPool() doesn't rebuild the pool - it just returns the cached pool that was built by rebuildCertPool()